### ISSUE TITLE
bundler from bundlerEnv.wrappedRuby fails with uninitialized constant Bundler::FeatureFlag 

### ISSUE THREAD

**Issue by thefloweringash** [NixOS/nixpkgs#78242]
**Describe the bug**

When running `bin/bundler` from a `bundlerEnv.wrappedRuby`, it fails with the message

```
`<module:Bundler>': uninitialized constant Bundler::FeatureFlag (NameError)
```

and stack trace:
```
Traceback (most recent call last):
	16: from /nix/store/35y07jj9xsdyflgdj6cs943ci97m5y2q-ruby-2.7.0/bin/bundler:23:in `<main>'
	15: from /nix/store/35y07jj9xsdyflgdj6cs943ci97m5y2q-ruby-2.7.0/bin/bundler:23:in `load'
	14: from /nix/store/s490w5mlc4r74hqz9k2vsf1ws1r1k663-thor-1.0.1/lib/ruby/gems/2.7.0/gems/bundler-1.17.3/exe/bundler:4:in `<top (required)>'
	13: from /nix/store/s490w5mlc4r74hqz9k2vsf1ws1r1k663-thor-1.0.1/lib/ruby/gems/2.7.0/gems/bundler-1.17.3/exe/bundler:4:in `load'
	12: from /nix/store/s490w5mlc4r74hqz9k2vsf1ws1r1k663-thor-1.0.1/lib/ruby/gems/2.7.0/gems/bundler-1.17.3/exe/bundle:22:in `<top (required)>'
	11: from /nix/store/s490w5mlc4r74hqz9k2vsf1ws1r1k663-thor-1.0.1/lib/ruby/gems/2.7.0/gems/bundler-1.17.3/lib/bundler/friendly_errors.rb:124:in `with_friendly_errors'
	10: from /nix/store/s490w5mlc4r74hqz9k2vsf1ws1r1k663-thor-1.0.1/lib/ruby/gems/2.7.0/gems/bundler-1.17.3/exe/bundle:23:in `block in <top (required)>'
	 9: from /nix/store/35y07jj9xsdyflgdj6cs943ci97m5y2q-ruby-2.7.0/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:72:in `require'
	 8: from /nix/store/35y07jj9xsdyflgdj6cs943ci97m5y2q-ruby-2.7.0/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:72:in `require'
	 7: from /nix/store/s490w5mlc4r74hqz9k2vsf1ws1r1k663-thor-1.0.1/lib/ruby/gems/2.7.0/gems/bundler-1.17.3/lib/bundler/cli.rb:6:in `<top (required)>'
	 6: from /nix/store/s490w5mlc4r74hqz9k2vsf1ws1r1k663-thor-1.0.1/lib/ruby/gems/2.7.0/gems/bundler-1.17.3/lib/bundler/cli.rb:7:in `<module:Bundler>'
	 5: from /nix/store/s490w5mlc4r74hqz9k2vsf1ws1r1k663-thor-1.0.1/lib/ruby/gems/2.7.0/gems/bundler-1.17.3/lib/bundler/cli.rb:97:in `<class:CLI>'
	 4: from /nix/store/s490w5mlc4r74hqz9k2vsf1ws1r1k663-thor-1.0.1/lib/ruby/gems/2.7.0/gems/bundler-1.17.3/lib/bundler.rb:478:in `feature_flag'
	 3: from /nix/store/35y07jj9xsdyflgdj6cs943ci97m5y2q-ruby-2.7.0/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:72:in `require'
	 2: from /nix/store/35y07jj9xsdyflgdj6cs943ci97m5y2q-ruby-2.7.0/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:72:in `require'
	 1: from /nix/store/s490w5mlc4r74hqz9k2vsf1ws1r1k663-thor-1.0.1/lib/ruby/gems/2.7.0/gems/bundler-1.17.3/lib/bundler/feature_flag.rb:3:in `<top (required)>'
/nix/store/s490w5mlc4r74hqz9k2vsf1ws1r1k663-thor-1.0.1/lib/ruby/gems/2.7.0/gems/bundler-1.17.3/lib/bundler/feature_flag.rb:4:in `<module:Bundler>': uninitialized constant Bundler::FeatureFlag (NameError)

```

**To Reproduce**

In the context of the [sample gist][], run 

```
$(nix-build --no-out-link test.nix --argstr rubyAttr ruby_2_7 -A wrappedRuby)/bin/bundler exec true
```

This seems to be new in `ruby_2_7`, running the following with `ruby_2_6` works as expected

```
$(nix-build --no-out-link test.nix --argstr rubyAttr ruby_2_6 -A wrappedRuby)/bin/bundler exec true
```

[sample gist]: https://gist.github.com/thefloweringash/9882f775aadfa7b16daee58091bfadc7

**Metadata**
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.13, NixOS, 20.03pre210072.8500d560a44 (Markhor)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.2`
 - channels(lorne): `""`
 - channels(root): `"nixos-20.03pre210072.8500d560a44"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: bundlerEnv
# a list of nixos modules affected by the problem
module:
```

**Comment by aanderse** [comment]
This is happening [with redmine](https://github.com/NixOS/nixpkgs/issues/80061) now too.

**Comment by aanderse** [comment]
If memory serves me correctly @alyssais might have some knowledge of the `ruby` infrastructure.

**Comment by aanderse** [comment]
@thefloweringash do you know many `ruby` applications are impacted by this?

cc @NixOS/nixos-release-managers

**Comment by zimbatm** [comment]
First digging:

The latest version of bundler is 2.1.4 and we are shipping 1.17.3: https://rubygems.org/gems/bundler.
It looks like the FeatureFlag was first introduced in v1.14.0 so it should at least exist: https://github.com/rubygems/bundler/commit/4a787997c6fb9b520251cd10ce0f2bda4e1c0e64#diff-1ae3342249cf56a91d8859833ca08947

**Comment by thefloweringash** [comment]
> @thefloweringash do you know many `ruby` applications are impacted by this?

I haven't found the time to take a good look at this yet sorry. I ran into it trying to upgrade a project from 2.6 to 2.7. I see in your stack trace for redmine that it, or a very similar issue, also affects 2.5.

**Comment by manveru** [comment]
Something must've changed in the way Bundler is required... I'll try to find some time today to dig into this, but I think it's safe to upgrade it to the latest version anyway.

**Comment by aanderse** [comment]
Any new information on this? Did you get a chance to look into it @manveru?

**Comment by manveru** [comment]
PR for fixing this specific issue is https://github.com/NixOS/nixpkgs/pull/81442 but there will be more work to upgrade all the lockfiles to the new version as well.

**Comment by aanderse** [comment]
Thank you very much to everyone who has put time and energy into this so far!

Does anyone have any updates on this? I'm really hoping we can resolve this before `20.03`...

**Comment by manveru** [comment]
Just a quick update: these issues should be fixed by https://github.com/NixOS/nixpkgs/pull/81442 but I need more people to actually try it.
If nobody complains, this will most likely be merged as is into 20.03.

**Comment by manveru** [comment]
https://github.com/NixOS/nixpkgs/pull/81442 is merged, so we can close this as well.
